lib/elixir/lib/exception.ex

```diff
@@ -163,48 +172,29 @@ defmodule Exception do
 
   This operation is potentially expensive, as it reads data
   from the filesystem, parse beam files, evaluates code and
-  so on. Currently the following exceptions may be annotated:
-
-    * `FunctionClauseError` - annotated with the arguments
-      used on the call and available clauses
+  so on.
 
+  If the exception module implements the optional `c:blame/2`
+  callbak, it will be invoked to perform the computation.
   """
   @spec blame(:error, any, stacktrace) :: {t, stacktrace}
   @spec blame(non_error_kind, payload, stacktrace) :: {payload, stacktrace} when payload: var
   def blame(kind, error, stacktrace)
 
   def blame(:error, error, stacktrace) do
-    case normalize(:error, error, stacktrace) do
-      %{__struct__: FunctionClauseError} = struct ->
-        blame_function_clause_error(struct, stacktrace)
+    %module{} = struct = normalize(:error, error, stacktrace)
 
-      _ ->
-        {error, stacktrace}
+    if Code.ensure_loaded?(module) and function_exported?(module, :blame, 2) do
+      module.blame(struct, stacktrace)
```

Comment:
Should we do some basic sanity check against the returned value to see, if it's valid?

Reply:
Let's go simple for now, I think we are fine.